ott/core/potentials.py

```diff
@@ -0,0 +1,161 @@
+from typing import Any, Callable, Dict, Sequence, Tuple
+
+import jax
+import jax.numpy as jnp
+import jax.scipy as jsp
+import jax.tree_util as jtu
+from typing_extensions import Literal
+
+from ott.geometry import pointcloud
+
+__all__ = ["DualPotentials", "EntropicPotentials"]
+Potential_t = Callable[[jnp.ndarray], float]
+
+
+@jtu.register_pytree_node_class
+class DualPotentials:
+  r"""The Kantorovich dual potentials :math:`f` and :math:`g`.
+
+  :math:`\nabla g` transports the source distribution to the target distribution
+  and :math:`\nabla f` the target distribution to the source distribution.
+
+  Args:
+    f: The first dual potential function.
+    g: The second dual potential function.
+  """
+
+  def __init__(self, f: Potential_t, g: Potential_t):
+    self._f = f
+    self._g = g
+
+  def transport(self, vec: jnp.ndarray, forward: bool = True) -> jnp.ndarray:
+    """Transport points using the dual potentials :attr:`f` and :attr:`g`.
+
+    Args:
```

Comment:
I think it's fine to call this transport but this won't be always true. 

Maybe

Transport `vec` using dual potentials according to Brenier formula.

Reply:
Done.